### PR TITLE
[CH11185] Disable dispatch of tracking events by default

### PR DESCRIPTION
**Pull request by sblaurock**
✅  All tests pass (225/225)

- `sendTrackingEvents` is now defaulted to false within request queue, meaning we will not dispatch events from the library when default options are supplied

- Changed logic so that events can still be queued (stored in localstorage) but will never be sent if `sendTrackingEvents` is not true

- Moved options relating to request queue into an options object rather than having them as individual options

- Updated all tests

Easier to view diff with [?w=1](https://github.com/Constructor-io/constructorio-client-javascript/pull/59/files?w=1)

### CODE REVIEW

File: src/utils/request-queue.js [hunk 1]
```diff
@@ -15,21 +15,25 @@ class RequestQueue {
...
-    this.send();
+    if (this.sendTrackingEvents) {
+      this.send();
+    }
   }
 
   // Add request to queue to be dispatched
   queue(url, method = 'GET', body) {
-    if (this.sendTrackingEvents && !this.humanity.isBot()) {
+    if (!this.humanity.isBot()) {
```
Comment (rubikzube) [inline src/utils/request-queue.js:36]: There may be legal implications here that are problematic... we should probably revert this change

File: src/utils/request-queue.js [hunk 2]
```diff
@@ -43,87 +47,93 @@ class RequestQueue {
...
-            body: JSON.stringify(nextInQueue.body),
-            mode: 'cors',
-            headers: { 'Content-Type': 'text/plain' },
-          });
-        }
-
-        if (request) {
-          this.requestPending = true;
-          const instance = this;
+    if (this.sendTrackingEvents) {
```
Comment (rubikzube) [inline src/utils/request-queue.js:50]: There may be legal implications here that are problematic... we should probably revert this change

File: src/constructorio.js [hunk 3]
```diff
@@ -47,8 +48,7 @@ class ConstructorIO {
       sessionId,
       userId,
       fetch,
-      trackingSendDelay,
-      sendTrackingEvents,
+      requestQueue,
```
Comment (rubikzube) [inline src/constructorio.js:51]: If we are aiming for speed, let's keep the existing interface as changing the name will result in changes in other repos also.  If we are aiming for correctness, maybe rename to `requestQueue` to `trackingOptions`?

**Review by rubikzube — CHANGES_REQUESTED** [review]

File: src/utils/request-queue.js (hunk 1 above)
Comment (sblaurock) [reply src/utils/request-queue.js]: > There may be legal implications here that are problematic... we should probably revert this change

Resolved

File: src/constructorio.js (hunk 3 above)
Comment (sblaurock) [reply src/constructorio.js:51]: > If we are aiming for speed, let's keep the existing interface as changing the name will result in changes in other repos also. If we are aiming for correctness, maybe rename to `requestQueue` to `trackingOptions`?

I figure we need to make a change anyway to explicitly turn on `sendTrackingEvents`, so it would be a good time to do this refactor now.

>  If we are aiming for correctness, maybe rename to requestQueue to trackingOptions

The event emitter options are currently named after the class - was following the same pattern. trackingOptions works for me, though it's a bit odd that idOptions is separate from this.

**Review by jjl014 — APPROVED** [review]
nice, lookin' good! 🙂

**Review by rubikzube — APPROVED** [review]
Let's do it/
